### PR TITLE
Improve IO lib exception and usage

### PR DESCRIPTION
**Pull request by PerfectSlayer**
Hi topjohnwu 

As for the previous release, here are some commits to try to improve the IO lib giving more details exceptions and ensuring some resources are properly closed.
Commits are independent so feel free to cherry-pick only the ones you like.

Regards

### CODE REVIEW

File: io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java [hunk 1]
```diff
@@ -74,16 +73,17 @@ class FifoOutputStream extends FilterOutputStream {
...
+            throw new FileNotFoundException(file.getAbsolutePath() + " is not a file but a directory");
         if (file.isBlock() || file.isCharacter()) {
             append = false;
-            return true;
         }
-        if (append)
-            return file.canWrite() || file.createNewFile();
-        return file.clear();
+        if (append && !file.canWrite() && !file.createNewFile()) {
+            throw new FileNotFoundException("Can write to file "+file.getAbsolutePath());
```
Comment (topjohnwu) [inline io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java:83]: Can you add spaces between the strings and `+`?

File: io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java [hunk 2]
```diff
@@ -74,16 +73,17 @@ class FifoOutputStream extends FilterOutputStream {
...
+        } else if (!file.clear()) {
+            throw new FileNotFoundException("Failed to clear file "+file.getAbsolutePath());
```
Comment (topjohnwu) [inline io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java:85]: Same as above

File: io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java (hunk 1 above)
Comment (PerfectSlayer) [reply io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java]: Sure, it's done.
(I avoid calling IDE formatting feature on code base that is not mine)

File: io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java (hunk 2 above)
Comment (PerfectSlayer) [reply io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java]: Same

**Review by topjohnwu — APPROVED** [review]

**Comment by PerfectSlayer** [comment]
That was fast! Thanks 🙏

**Pull request merged**